database/index.go

```diff
@@ -136,25 +183,88 @@ func (idx *Index) Delete(v document.Value, k []byte) error {
 	return engine.ErrKeyNotFound
 }
 
+// validatePivots returns an error when the pivots are unsuitable for the index:
+// - no pivots at all
+// - having pivots length superior to the index arity
+// - having the first pivot without a value when the subsequent ones do have values
+func (idx *Index) validatePivots(pivots []document.Value) {
```

Comment:
This shows that we might need a type for the pivot itself, and for the list of pivots as well, I think it will help understanding how pivots should work and separate pivot code from index code.

Something like this:
```go
type Pivot document.Value

func (p *Pivot) Validate() error

type Pivots []document.Value

func (p *Pivots) Validate() error
```